pcdsutils/json_to_table.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+"""
+Take in a JSON document with an array at the top-level and generate a
+PrettyTable from it.
+"""
+
+import argparse
+import dataclasses
+import enum
+import json
+import sys
+from typing import Any, BinaryIO, List, Optional, cast
+
+import prettytable
+
+
+class TableStyle(enum.Enum):
+    """Table style/format to print."""
+    html = enum.auto()
+    csv = enum.auto()
+    latex = enum.auto()
+
+    default = enum.auto()
+    msword_friendly = enum.auto()
+    plain_columns = enum.auto()
+    markdown = enum.auto()
+    orgmode = enum.auto()
+    double_border = enum.auto()
+    # random = enum.auto()
+
+
+@dataclasses.dataclass
+class ProgramArguments:
+    """Argparse arguments for json_to_table."""
+    input_file: BinaryIO
+    sort_key: str
+    columns: List[str]
+    format: str = "markdown"
+    list_delimiter: str = ", "
+    max_col_width: int = 80
+
+
+def string_for_table(
+    value: Any, max_length: int = 80, list_delimiter: str = ", "
+) -> str:
+    """Fix the value for display in the table."""
+    if isinstance(value, list):
+        result = list_delimiter.join(string_for_table(v) for v in value)
+    else:
+        result = str(value)
+
+    if len(result) > max_length:
+        return result[:max_length] + "..."
```

Comment:
Do we want to truncate at `max_length - 3` so that the lines with and without ellipses are aligned?

Reply:
Good call 👍

Reply:
Resolved in dd14ca1